src/extension.ts

```diff
@@ -1,98 +1,175 @@
-import * as vscode from 'vscode';
-
-import { TelemetryEvent } from './TelemetryEvent';
-import { TelemetryEventQueue } from './TelemetryEventQueue';
-import { Reporter } from './Reporter';
-import { Logger } from './Logger';
-
-const OPTIN_REQUESTED_KEY = 'redhat.telemetry.optInRequested';
-
+import * as vscode from "vscode";
+import { SegmentInitializer } from "./segmentInitializer";
+import { Logger } from "./utils/logger";
+import Analytics from "analytics-node";
+import { TelemetryEventQueue } from "./utils/telemetryEventQueue";
+import { TelemetryEvent } from "./interfaces/telemetryEvent";
+import { Reporter } from "./reporter";
+
+/* 
+  OPT_IN_STATUS == "true" if user Agreed
+  false when denied for telemetry data collection     
+*/
+const OPT_IN_STATUS = "redhat.telemetry.optInRequested";
+
+// this method is called when your extension is activated
 export function activate(context: vscode.ExtensionContext) {
-  Reporter.initialize(context);
+  Logger.log('"vscode-commons" is now active!');
+
+  //  instance of initialized segment object
+  const analytics: Analytics | undefined = SegmentInitializer.initialize(
+    context
+  );
 
   if (!getTelemetryEnabledConfig()) {
-    Logger.log('redhat.telemetry.enabled is false');
+    Logger.log("redhat.telemetry.enabled is false");
     TelemetryEventQueue.initialize();
   } else {
-    Logger.log('redhat.telemetry.enabled is true');
+    Logger.log("redhat.telemetry.enabled is true");
   }
 
   openTelemetryOptInDialogIfNeeded(context);
 
-  context.subscriptions.push(onDidChangeTelemetryEnabled());
+  context.subscriptions.push(
+    vscode.commands.registerCommand("vscodeCommons.openWebPage", openWebPage)
+  );
 
-  // This command exists only for testing purposes. Should delete later.
-  context.subscriptions.push(vscode.commands.registerCommand('vscodeCommons.clearStateAndSettings', () => {
```

Comment:
we might want to keep that test command for now, for testing purposes, remove it later once we're getting closer to  releasing

Reply:
Sure, I have added them back.